src/lib.rs

```diff
@@ -319,10 +321,13 @@ fn do_write(mut remaining: &[u8]) {
             grant[offset..][..min].copy_from_slice(&remaining[..min]);
             offset += min;
 
+            BBQ_GRANT_W.offset.store(offset, Ordering::Relaxed);
+
             remaining = &remaining[min..];
 
             if offset >= glen {
                 grant.commit(offset);
+                BBQ_STATE.store(logstate::INIT_IDLE, Ordering::Release);
```

Comment:
This line is handled in the `take()` function here: https://github.com/jamesmunns/defmt-bbq/blob/abb45806a8015d0280278057e32453c25430cb5c/src/lib.rs#L171-L195